frontend/plan/components/meetUtil.tsx

```diff
@@ -38,14 +46,17 @@ export const getConflictGroups = (meetings) => {
     // https://en.wikipedia.org/wiki/Disjoint-set_data_structure
     // meetings m1 and m2 are in the same conflict set if m1 and m2 conflict
     // with at least one meeting m3 which is also in the set (m3 can be m1 or m2).
-    const conflicts = {};
-    const merge = (m1, m2) => {
-        conflicts[m2.id] = new Set([...conflicts[m1.id], ...conflicts[m2.id]]);
-        conflicts[m1.id] = conflicts[m2.id];
+    const conflicts: { [index: number]: Set<MeetingBlock> } = {};
+    const merge = (m1: MeetingBlock, m2: MeetingBlock) => {
+        conflicts[m2.id as number] = new Set([
+            ...Array.from(conflicts[m1.id as number]),
+            ...Array.from(conflicts[m2.id as number]),
+        ]);
+        conflicts[m1.id as number] = conflicts[m2.id as number];
     };
 
     meetings.forEach((m) => {
-        conflicts[m.id] = new Set([m]);
+        conflicts[m.id as number] = new Set([m]);
```

Comment:
wondering why we need to cast the `id` to be a `number` everywhere? looks like it's defined as a number in the types file. Is it because it may be undefined? If so, I know it's annoying, but to fully take advantage of the type system we should do an `if (m.id)` before doing anything with the `id` property.